ktor-io/posix/src/io/ktor/utils/io/ConditionNative.kt

```diff
@@ -6,7 +6,7 @@ import kotlinx.coroutines.*
 import kotlin.coroutines.*
 
 internal actual class Condition actual constructor(val predicate: () -> Boolean) {
-    private var continuation: Continuation<Unit>? by shared(null)
+    private var continuation: Continuation<Unit>? = null
```

Comment:
Looks like `Condition` class is almost the same on all platforms. Can it be commonized now with new MM?

Reply:
I'm not sure if it's used anymore, so maybe we should just drop it